### ISSUE TITLE
ShellCheckRule errors on custom shell `echo {0}`

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#775]
### Version

GH-Lint (ghlint) version 0.5.1-SNAPSHOT, local CLI.

### Reproduction steps

1. Create `.github/workflows/x.yml`:

   ```yaml
   name: Repro

   on: push

   jobs:
     repro:
       runs-on: ubuntu-latest
       steps:
         - name: Custom shell
           shell: echo {0}
           run: Hello
   ```

2. Run:

   ```shell
   ghlint --exit .github/workflows/x.yml
   ```

### Expected behavior

`ShellCheckRule` should skip unsupported custom shells (or otherwise report a normal finding), without failing the rule execution.

### Actual behavior

GH-Lint reports an internal rule error:

```text
RuleErrored at .github/workflows/x.yml:...: net.twisterrob.ghlint.rules.ShellCheckRule errored while checking .github/workflows/x.yml:
java.lang.IllegalStateException: Error running command: Unknown shell: echo {0}
    at net.twisterrob.ghlint.rules.shell.ShellCheck$Companion.runCommand(SourceFile:88)
    at net.twisterrob.ghlint.rules.shell.ShellCheck.check(SourceFile:40)
    at net.twisterrob.ghlint.rules.ShellCheckRule.visitWorkflowRunStep(SourceFile:23)
```

### Additional Context

#436 covers a `DuplicateShellOnSteps` false positive involving the same custom shell syntax, but not this `ShellCheckRule` crash.
